### PR TITLE
Feature/region

### PR DESCRIPTION
**Pull request by lbebber**
## What's the purpose of this pull request?

Adds region context to search queries.

Still WIP, but merging into `next` to avoid big rebases

Also passes down regionId to PDP

Can be seen working on https://sfj-2d37e44--carrefourbrfood.vtex.app/

<em>Considering the context, what is the problem we'll solve? Where in VTEX's big picture our issue fits in? Write a tweet about the context and the problem itself.</em>

## How it works? 
<em>Tell us the role of the new feature, or component, in its context.</em>

## How to test it?
<em>Describe the steps with bullet points. Is there any external link that can be used to better test it or an example?</em> 

## References
<em>Spread the knowledge: is there any content you used to create this PR that is worth sharing?</em>  

<em>Extra tip: adding references to related issues or mentioning people important to this PR may be good for the documentation and reviewing process</em>

### CODE REVIEW

File: packages/gatsby-theme-store/src/components/Region/index.tsx [hunk 1]
```diff
@@ -0,0 +1,22 @@
+/** @jsx jsx */
```
Comment (lbebber) [inline packages/gatsby-theme-store/src/components/Region/index.tsx:1]: Dummy component for debugging, will be removed later

File: packages/gatsby-theme-store/src/sdk/search/useSearchFiltersFromPageContext.ts [hunk 2]
```diff
@@ -98,13 +97,5 @@ export const useSearchFiltersFromPageContext = (
...
-  }, [
-    search,
-    pageContext.staticPath,
-    pageContext.query,
-    pageContext.map,
-    pageContext.orderBy,
-    pageContext.selectedFacets,
-    pathname,
-  ])
+  }, [search, pageContext, pathname])
```
Comment (klzns) [inline packages/gatsby-theme-store/src/sdk/search/useSearchFiltersFromPageContext.ts:100]: Is pageContext memoized? If it's not, it will run everytime.

Comment (lbebber) [reply packages/gatsby-theme-store/src/sdk/search/useSearchFiltersFromPageContext.ts:100]: hmmm this was an eslint autofix. I trusted it knew what it was doing, but I'm gonna revert to be sure

File: packages/gatsby-theme-store/src/sdk/search/useSearchFiltersFromPageContext.ts [hunk 3]
```diff
@@ -58,7 +58,11 @@ export const useSearchFiltersFromPageContext = (
 
   return useMemo(() => {
     const params = new URLSearchParams(search)
-    let { query, map } = pageContext
+    // Quick fix for some questions regarding memo dependencies
+    /* eslint-disable-next-line prefer-destructuring */
+    let query = pageContext.query
+    /* eslint-disable-next-line prefer-destructuring */
+    let map = pageContext.map
```
Comment (klzns) [inline packages/gatsby-theme-store/src/sdk/search/useSearchFiltersFromPageContext.ts:65]: You can fix this eslint ignore with:

```tsx
const { query, staticPath, query, map, orderBy, selectedFacets } = pageContext

useMemo(() => {


}, [search, staticPath, query, map, orderBy, selectedFacets, pathname])
```

**Review by klzns — APPROVED** [review]

**Review by emersonlaurentino — APPROVED** [review]

File: packages/gatsby-theme-store/src/sdk/search/useSearchFiltersFromPageContext.ts (hunk 3 above)
Comment (lbebber) [reply packages/gatsby-theme-store/src/sdk/search/useSearchFiltersFromPageContext.ts:65]: @klzns  Good idea, but I'm just gonna merge for now and do this in a future PR

**Pull request merged**